frontend/src/types.ts

```diff
@@ -635,6 +635,27 @@ export const ProgressionStepTitles = {
   [ProgressionStep.SERVER_STARTED]: 'Server started',
 };
 
+export const AssociatedSteps = {
+  [ProgressionStep.NOTEBOOK_CONTAINER_STARTED]: [
+    ProgressionStep.SERVER_REQUESTED,
+    ProgressionStep.POD_CREATED,
+    ProgressionStep.POD_ASSIGNED,
+    ProgressionStep.PVC_ATTACHED,
+    ProgressionStep.INTERFACE_ADDED,
+    ProgressionStep.PULLING_NOTEBOOK_IMAGE,
+    ProgressionStep.NOTEBOOK_IMAGE_PULLED,
+    ProgressionStep.NOTEBOOK_CONTAINER_CREATED,
+  ],
+  [ProgressionStep.OAUTH_CONTAINER_STARTED]: [
+    ProgressionStep.PULLING_OAUTH,
+    ProgressionStep.OAUTH_PULLED,
+    ProgressionStep.OAUTH_CONTAINER_CREATED,
+    ProgressionStep.OAUTH_CONTAINER_STARTED,
+  ],
+  [ProgressionStep.POD_ASSIGNED]: [ProgressionStep.POD_CREATED],
+  [ProgressionStep.SERVER_STARTED]: [ProgressionStep],
```

Comment:
`[ProgressionStep]` is an array wrapping an object (the enum) -- you won't be able to match items in the `progressStep[].step` against this as they will be a string

```suggestion
  [ProgressionStep.SERVER_STARTED]: Object.values(ProgressionStep),
```
This will give you a "dynamic" list of all items -- ie "when I hit Server started, everything is success" -- it'll catch itself, but that's not really a problem, it'll already be success and there is no harm setting it to the same value.